File: FCSeafood/FCSeafood.BusinessObjects/Models/Events/ItemModel.cs
namespace FCSeafood.BusinessObjects.Models.Events;

public class ItemModel {
    public Guid Id { get; set; }

    public string Name { get; set; } = "";

    public string Code { get; set; } = "";
    public string ImagePath { get; set; } = "";

    public double Price { get; set; }

    public CategoryTModel Category { get; set; } = new();

    public SubcategoryTModel Subcategory { get; set; } = new();

    public ItemStatusTModel ItemStatus { get; set; } = new();

    public double QuantityPerKg { get; set; }

    public int LikeCount { get; set; }

    public int DislikeCount { get; set; }

    public string? Brand { get; set; }

    public string? Type { get; set; }

    public string? Finishing { get; set; }

    public bool IsPackaging { get; set; }

    public double? FatsPer100Gram { get; set; }

    public double? CarbohydratesPer100Gram { get; set; }

    public double? ProteinsPer100Gram { get; set; }

    public double? KcalPer100Gram { get; set; }

    public double? HumidityPerPercent { get; set; }

    public DateTime ExpirationDate { get; set; }

    public double TemperatureStorage { get; set; }

    public TemperatureUnitTModel TemperatureUnit { get; set; } = new();

    public string? Description { get; set; }
}

File: FCSeafood-Angular/src/app/common/models/item.model.ts
import { CategoryTModel } from "@common-models/category-type.model";
import { SubcategoryTModel } from "@common-models/subcategory-type.model";
import { ItemStatusTModel } from "@common-models/item-status-type.model";
import { TemperatureUnitTModel } from "@common-models/temperature-unit-type.model";

export class ItemModel {
  id: string = ''; // guid
  name: string = '';
  code: string = '';
  imagePath: string = '';
  price: number = 0.0;
  category: CategoryTModel = new CategoryTModel();
  subcategory: SubcategoryTModel = new SubcategoryTModel();
  itemStatus: ItemStatusTModel = new ItemStatusTModel();
  quantityPerKg: number = 0.0;
  likeCount: number = 0.0;
  dislikeCount: number = 0.0;
  brand: string | null = null;
  type: string | null = null;
  finishing: string | null = null;
  isPackaging: boolean = true;
  fatsPer100Gram: number | null = null;
  carbohydratesPer100Gram: number | null = null;
  proteinsPer100Gram: number | null = null;
  kcalPer100Gram: number | null = null;
  humidityPerPercent: number | null = null;
  expirationDate: Date = new Date();
  temperatureStorage: number = 0.0;
  temperatureUnit: TemperatureUnitTModel = new TemperatureUnitTModel();
  description: string | null = null;
}
